### DISCUSSION TITLE
wezterm is too slow

### DISCUSSION THREAD

**Discussion by og900aero** [wezterm/wezterm#4616] [General]
I try wezterm, and I find that it is slower than average. The opening speed is also slower even on terminals that are old.

xterm: 0.101s
urxvt: 0.111s 
st: 0.090s
alacritty: 0.092s
tilix: 0.089s
**wezterm: 0.347s**

command: ls -laR /

xterm: 52.769s
urxvt: 16.579s 
st: 16.623s
alacritty: 15.444s
tilix: 24.228s
**wezterm: 22.045s**

Why is it so slow? Even the opening speed is three times slower than others, which is already quite visible. Is it because of the Rust language, or do I need some settings that will make it have a more normal speed?

**Comment by bew** [comment]
For `ls -laR /` you can read about this in https://github.com/wez/wezterm/issues/1453#issuecomment-1000068807

For opening speed, you'll have to provide more information on your environment / OS / version / ...
Also check existing issues about this, see if your slowness comes from the same reasons

**Comment by og900aero** [comment]
> For `ls -laR /` you can read about this in [#1453 (comment)](https://github.com/wez/wezterm/issues/1453#issuecomment-1000068807)
> 
> For opening speed, you'll have to provide more information on your environment / OS / version / ... Also check existing issues about this, see if your slowness comes from the same reasons

I use Debian 12 and wezterm 20231124-202004-f07cc837

My config file:

```
local act = wezterm.action

-- This table will hold the configuration.
local config = {}

-- In newer versions of wezterm, use the config_builder which will
-- help provide clearer error messages
if wezterm.config_builder then
  config = wezterm.config_builder()
end

-- This is where you actually apply your config choices
config.enable_wayland = false
config.default_cwd = "/home/Data/"
config.check_for_updates = true
config.show_update_window = true
config.check_for_updates_interval_seconds = 604800 -- weekly
config.scrollback_lines = 500
-- config.color_scheme = 'AdventureTime'
-- config.color_scheme = 'Colors (base16)'
-- config.color_scheme = 'Afterglow'
config.color_scheme = 'Google Dark (Gogh)'
-- config.enable_tab_bar = false
config.hide_tab_bar_if_only_one_tab = true
config.font = wezterm.font('IosevkaTerm Nerd Font Mono', { weight = 'Regular', italic = false })
config.font_size = 14.0
config.line_height = 1.0
config.window_close_confirmation = 'NeverPrompt'
config.disable_default_key_bindings = true

wezterm.action.SpawnCommandInNewWindow {
  label = 'List all the files!',
  args = { 'ls', '-la' },
  cwd = '/home/Data',
}

config.keys = {
  {
    key = 'P',
    mods = 'CTRL',
    action = wezterm.action.ActivateCommandPalette,
  },
}

config.mouse_bindings = {
	{
		event = { Down = { streak = 1, button = "Right" } },
		mods = "NONE",
		action = wezterm.action_callback(function(window, pane)
			local has_selection = window:get_selection_text_for_pane(pane) ~= ""
			if has_selection then
				window:perform_action(act.CopyTo("ClipboardAndPrimarySelection"), pane)
				window:perform_action(act.ClearSelection, pane)
			else
				window:perform_action(act({ PasteFrom = "Clipboard" }), pane)
			end
		end),
	},
}

-- and finally, return the configuration to wezterm
return config

```

**Reply by vollink** [reply]
Just another USER here, but boy, it sure seems like the startup of a LUA sub-library as well as a full interpretation of a configuration file by that interpreter (which is something none of those other terminals do) would account for the fraction of a second you are asking about.  It might be more useful to start this without any configuration, and see how those numbers differ.

**Comment by og900aero** [comment]
> Just another USER here, but boy, it sure seems like the startup of a LUA sub-library as well as a full interpretation of a configuration file by that interpreter (which is something none of those other terminals do) would account for the fraction of a second you are asking about. It might be more useful to start this without any configuration, and see how those numbers differ.

Withouit config file: 0.222s

Better, but slower than other terminals.

**Reply by wez** [reply]
Take a look through https://github.com/wez/wezterm/issues?q=slow+xrandr to see if perhaps this is a system driver/kernel configuration issue that manifests with xrandr.

**Comment by og900aero** [comment]
> Take a look through https://github.com/wez/wezterm/issues?q=slow+xrandr to see if perhaps this is a system driver/kernel configuration issue that manifests with xrandr.

I looked at the links, but xrandr runs quickly for me, there are no special drivers installed, no video card driver, or anything. Standard drivers are installed:
- xserver-xorg-video-intel
- intel-media-va-driver
- xserver-xorg-video-nouveau

I tried that if I activate this line:
`config.color_scheme = 'Google Dark (Gogh)'`

Then the startup time time will double.
But if I take everything out, it's still slow compare others. If the wezterm is slow even with the basic Linux drivers, then unfortunately this is lost for me, because all other terminals can handle it properly. Thanks for the help.
